File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.java
class Solution
{
    public int solution(int [][]board)
    {
        int n = board.length;
        int m = board[0].length;
        if(n==1 && m==1 && board[0][0]==1) return 1;
        int answer = Integer.MIN_VALUE;
        for(int i=1; i<n; i++) {
            for(int j=1; j<m; j++) {
                if(board[i][j] == 1) {
                    int temp = Math.min(board[i-1][j], board[i][j-1]);
                    temp = Math.min(temp, board[i-1][j-1]);
                    board[i][j] = temp+1;
                    answer = Math.max(board[i][j], answer);
                }
            }
        }


        return answer*answer;
    }
}

File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.cpp
#include <iostream>
#include<vector>
#include<climits>

using namespace std;

int solution(vector<vector<int>> board)
{
    int n = board.size();
    int m = board[0].size();
    
    vector<vector<int>> dp(n+1, vector<int>(m+1, 0));
    int answer = INT_MIN;
    
    for(int i=1; i<n+1; i++) {
        for(int j=1; j<m+1; j++) {
            //최솟값 찾기
            int temp = dp[i-1][j];
            if(temp > dp[i][j-1]) temp = dp[i][j-1];
            if(temp > dp[i-1][j-1]) temp = dp[i-1][j-1];
            
            if(board[i-1][j-1] == 1) {
                dp[i][j] = temp + 1;
                if(answer < dp[i][j]) answer = dp[i][j];
            }
        }
    }
    
    if(answer == INT_MIN) return 0;
    else return (answer*answer);

}